libs/functions/calculate_projections.py

```diff
@@ -0,0 +1,99 @@
+import pandas as pd
+import datetime
+import os.path
+import simplejson
+
+from libs.us_state_abbrev import us_state_abbrev
+from libs.datasets import FIPSPopulation
+from libs.datasets.model_output_schema import MODEL_OUTPUT_SCHEMA, MODEL_OUTPUT_SCHEMA_EXCLUDED_COLUMNS
+from libs.datasets.projections_schema import CALCULATED_PROJECTION_HEADERS_STATES, CALCULATED_PROJECTION_HEADERS_COUNTIES
+
+def _calc_short_fall(x):
+    return abs(x.beds - x.all_hospitalized) if x.all_hospitalized > x.beds else 0
+
+def _get_hospitals_and_shortfalls(df, date_out):
+    first_record_after_date = df[(df.date > date_out)].iloc[0]
+    hospital_short_fall_columns = ['all_hospitalized', 'short_fall']
+    return tuple(first_record_after_date[hospital_short_fall_columns].values)
+
+def _read_json_as_df(path):
+    # TODO: read this from a dataset class
+    df = pd.DataFrame.from_records(simplejson.load(open(path,'r')),
+    columns=MODEL_OUTPUT_SCHEMA, exclude=MODEL_OUTPUT_SCHEMA_EXCLUDED_COLUMNS)
+
+    df['date'] = pd.to_datetime(df.date)
+    df['all_hospitalized'] = df['all_hospitalized'].astype('int')
+    df['beds'] = df['beds'].astype('int')
+    df['dead'] = df['dead'].astype('int')
+    return df
+
+def _calculate_projection_data(file_path):
+    """
+    Given a file path, return the calculations we perform for that file. 
+    Note in the future maybe return a data type to keep type clarity
+    """
+    # get 16 and 32 days out from now
+    today = datetime.datetime.now()
+    sixteen_days = today + datetime.timedelta(days=16)
+    thirty_two_days = today + datetime.timedelta(days=32)
+
+    if os.path.exists(file_path):
```

Comment:
This path should exist right?

Reply:
nah i think we could have fips that don't have data for right?